interfaces/README.md

```diff
@@ -143,3 +143,95 @@ Run these commands from every app's own 'root'-folder.
+          expect(component.someValye).toBeFalsy();  // check for a variable to be FALSE
+        });
+
+#### Testing method callbacks and changes
+-  By utilizing the `spy` provided within the jasmine framework, we should always test and verify that the appropriate methods have been called.
+
+        it('some_method: should call another fn', () => {
+          spyOn(event, "preventDefault"); // Monitor the said method
+          component.doLogin(event); // call some_method
+          expect(event.preventDefault).toHaveBeenCalled(); // check for the monitored method to have been called
+        });
+
+#### Testing conditional statements
+-  Make separate `it` blocks for different conditions.
+
+        it("Test when xyz === 'somethinf'") ....
+        it("Test when xyz !== 'somethinf'") ....
+
+#### Testing conditional statements
+-  Make separate `it` blocks for different conditions.
+
+        it("Test when xyz === 'somethinf'") ....
+        it("Test when xyz !== 'somethinf'") ....
+
+
+#### Testing Async Mehthods (i.e. methods which make an API call)
+-  Make a Spy for the specific async call which returns a Promise object. For example a method containing a call routine `this.programsService.changePassword` can be spied using following
+
+        let spy = spyOn(component.programsService, "changePassword").and.returnValue(Promise.resolve(true));
+
+
+-  Based on the changes / executions upon the completion of the async request, we should aim to test the changes and modifications.
+
+        spy.calls.mostRecent().returnValue.then(() => { // block to test what happenes after the async calls.
+            // Here goes expectations and changes
+        });
+
+-  Make sure the `done()` method is used to account for the async calls and fake async stubs/spies.
+
+        it('XYZ', (done) => {
+          // spies and stubs
+
+          spy.calls.mostRecent().returnValue.then(() => {
+              // tests
+              done(); // to complete the tests
+          });
+        });
+
+#### Testing HTML Tags
+-  By using the `defaultEl` and the monitoring the changes within the HTML pages. However, the testing here does not bring a lot of productivity in terms of what we get out of it. So, we can choose to discard this aspect of testing.
+-  HTML tags are tested by matching the `string` values, which is not very intuitive with `i18n` modules in use
+
+        it('label changes / tag changes', (done) => {
+                fixture.detectChanges();
+                expect(el.nativeElement.textContent).toBe('Login'); // verify the element
+                method_call() // call the method which is being tested
+                fixture.detectChanges()
+                expect(el.nativeElement.textContent).toBe('Logout'); // verify the changes     
+        });
+
+NOTE: It isn't necessary to test all the variables and all method calls, however a highlight of what the method is supposed to accomplish should be reflected within the test cases.
```

Comment:
I think this sentence should be moved up a bit, somewhere around line 160-165...